### PR TITLE
New Italian dishes and recipes

### PR DESCRIPTION
**Pull request by Jackody2**
#### Summary
Content "new Italian dishes and recipes"

#### Purpose of change
add new Italian dishes and their respective recipes

#### Describe the solution
add new Italian dishes and their respective recipes

#### Testing
All tested locally

### CODE REVIEW

File: data/json/itemgroups/Food/food.json [hunk 1]
```diff
@@ -1020,4 +1026,4 @@
       { "item": "hamburger", "prob": 5 }
     ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/itemgroups/Food/food.json:1029]:
```suggestion
]

```

File: data/json/itemgroups/Food/irradiated.json [hunk 2]
```diff
@@ -96,7 +97,8 @@
       [ "irradiated_potato", 50 ],
       [ "irradiated_cucumber", 50 ],
       [ "irradiated_celery", 50 ],
-      [ "irradiated_rhubarb", 50 ]
+      [ "irradiated_rhubarb", 50 ],
+      [ "irradiated_eggplant", 50 ]
     ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/itemgroups/Food/irradiated.json:104]:
```suggestion
]

```

File: data/json/items/comestibles/casseroles.json [hunk 3]
```diff
@@ -74,5 +74,29 @@
...
+    "material": [ "veggy", "milk" ],
+    "primary_material": "veggy",
+    "volume": "200 ml",
+    "flags": [ "EDIBLE_FROZEN" ],
+    "charges": 4,
+    "vitamins": [ [ "calcium", 15 ], [ "iron", 2 ], [ "vitC", 7 ] ],
+    "fun": 8
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/casseroles.json:102]:
```suggestion
]

```

File: data/json/items/comestibles/dairy.json [hunk 4]
```diff
@@ -472,5 +472,72 @@
...
+    "material": [ "milk" ],
+    "primary_material": "cheese",
+    "volume": "180 ml",
+    "flags": [ "EDIBLE_FROZEN" ],
+    "charges": 4,
+    "fun": 2,
+    "vitamins": [ [ "calcium", 19 ], [ "iron", 5 ] ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/dairy.json:543]:
```suggestion
]

```

File: data/json/items/comestibles/irradiated_veggy.json [hunk 5]
```diff
@@ -152,5 +152,16 @@
...
+    "copy-from": "eggplant",
+    "name": { "str": "irradiated eggplant" },
+    "description": "An irradiated purple eggplant.",
+    "proportional": { "price": 2.0 },
+    "spoils_in": "0 hours",
+    "container": "bag_plastic",
+    "extend": { "flags": [ "RAD_STERILIZED_EDIBLE_FOREVER" ] }
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/irradiated_veggy.json:167]:
```suggestion
]

```

File: data/json/items/comestibles/junkfood.json [hunk 6]
```diff
@@ -1551,5 +1551,53 @@
...
+    "material": [ "wheat", "veggy" ],
+    "primary_material": "wheat",
+    "volume": "150 ml",
+    "flags": [ "EDIBLE_FROZEN" ],
+    "charges": 4,
+    "vitamins": [ [ "calcium", 10 ], [ "iron", 6 ] ],
+    "fun": 8
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/junkfood.json:1603]:
```suggestion
]

```

File: data/json/items/comestibles/raw_veggy.json [hunk 7]
```diff
@@ -1159,5 +1159,26 @@
...
+    "description": "A normal and healty purple eggplant.",
+    "healthy": -1,
+    "symbol": "e",
+    "looks_like": "eggplant",
+    "flags": [ "RAW" ],
+    "vitamins": [ [ "calcium", 9 ], [ "iron", 2 ], [ "vitC", 10 ] ],
+    "fun": -5
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/raw_veggy.json:1184]:
```suggestion
]

```

File: data/json/items/comestibles/seed.json [hunk 8]
```diff
@@ -929,5 +929,14 @@
...
+    "type": "COMESTIBLE",
+    "id": "seed_eggplant",
+    "copy-from": "seed",
+    "price": 70,
+    "name": { "str_sp": "eggplant seeds" },
+    "description": "Some eggplant seeds.",
+    "seed_data": { "plant_name": "eggplant", "fruit": "eggplant", "byproducts": [ "withered" ], "grow": "60 days" }
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/items/comestibles/seed.json:942]:
```suggestion
]

```

File: data/json/recipes/food/bread.json [hunk 9]
```diff
@@ -301,5 +301,57 @@
...
+      [ [ "yeast", 2 ] ],
+      [ [ "sugar", 4 ], [ "artificial_sweetener", 4 ] ],
+      [ [ "eggs_any_shape", 4, "LIST" ] ],
+      [ [ "any_butter_or_oil", 2, "LIST" ] ],
+      [ [ "salt", 2 ] ],
+      [ [ "milk", 2 ] ]
+    ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/recipes/food/bread.json:357]:
```suggestion
]

```

File: data/json/recipes/food/casseroles.json [hunk 10]
```diff
@@ -121,5 +121,28 @@
...
+      [ [ "eggplant", 5 ] ],
+      [ [ "salt", 8 ] ],
+      [ [ "seasoning_italian", 7 ], [ "wild_herbs", 7 ], [ "pepper", 7 ] ],
+      [ [ "any_butter_or_oil", 4, "LIST" ] ],
+      [ [ "cheese", 6 ], [ "can_cheese", 6 ], [ "mozzarella_cheese", 6 ] ],
+      [ [ "sauce_red", 4 ] ]
+    ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/recipes/food/casseroles.json:148]:
```suggestion
]

```

File: data/json/recipes/food/dairy_products.json [hunk 11]
```diff
@@ -192,5 +192,43 @@
...
+    "difficulty": 3,
+    "charges": 6,
+    "book_learn": [ [ "cookbook_italian", 3 ] ],
+    "time": "30 m",
+    "batch_time_factors": [ 70, 2 ],
+    "qualities": [ { "id": "CHURN", "level": 1 } ],
+    "components": [ [ [ "milk_cream", 8 ] ], [ [ "lemon", 2 ] ] ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/recipes/food/dairy_products.json:234]:
```suggestion
]

```

File: data/json/recipes/food/pasta.json [hunk 12]
```diff
@@ -222,5 +222,51 @@
...
+    "components": [
+      [ [ "spaghetti_raw", 1 ] ],
+      [ [ "water_clean", 1 ] ],
+      [ [ "sauce_red", 3 ] ],
+      [ [ "cheese_standard", 1, "LIST" ], [ "pecorino_cheese", 1 ] ],
+      [ [ "meat_cooked", 1, "LIST" ], [ "dry_meat", 1 ], [ "bacon", 1 ] ]
+    ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/recipes/food/pasta.json:272]:
```suggestion
]

```

File: data/json/recipes/food/seed.json [hunk 13]
```diff
@@ -345,5 +345,12 @@
...
+  },
+  {
+    "result": "seed_eggplant",
+    "type": "recipe",
+    "copy-from": "seed_extraction_base",
+    "components": [ [ [ "eggplant", 1 ] ] ],
+    "charges": 8
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/recipes/food/seed.json:356]:
```suggestion
]

```

File: data/json/items/comestibles/dairy.json [hunk 14]
```diff
@@ -472,5 +472,72 @@
...
+    "id": "mascarpone",
+    "name": { "str_sp": "mascarpone" },
+    "weight": "250 g",
+    "color": "white",
+    "spoils_in": "12 days",
+    "container": "cup_plastic",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 450,
+    "description": "a portion of mascarpone, a dairy product obtained from the processing of cream and citric acid, a typical Italian spreadable cheese.",
```
Comment (TheShadowFerret) [inline data/json/items/comestibles/dairy.json:511]:
```suggestion
    "description": "A portion of mascarpone, a dairy product obtained from the processing of cream and citric acid, a typical Italian spreadable cheese.",
```

File: data/json/items/comestibles/dairy.json [hunk 15]
```diff
@@ -472,5 +472,72 @@
...
+    "id": "pecorino_cheese",
+    "name": { "str_sp": "pecorino" },
+    "weight": "100 g",
+    "color": "white",
+    "spoils_in": "180 days",
+    "container": "wrapper",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 350,
+    "description": "a slice of pecorino, famous Italian hard cheese used in pasta carbonara.",
```
Comment (TheShadowFerret) [inline data/json/items/comestibles/dairy.json:532]:
```suggestion
    "description": "A slice of pecorino, famous Italian hard cheese used in pasta carbonara.",
```
~~I use parmesan cuz I cheap~~

File: data/json/items/comestibles/junkfood.json [hunk 16]
```diff
@@ -1551,5 +1551,53 @@
...
+    "weight": "150 g",
+    "color": "yellow",
+    "spoils_in": "21 days",
+    "container": "pan",
+    "comestible_type": "FOOD",
+    "symbol": "c",
+    "quench": -2,
+    "healthy": -1,
+    "calories": 565,
+    "description": "a pan containing pasta carbonara, a famous Italian dish made with egg, pecorino and bacon.",
```
Comment (TheShadowFerret) [inline data/json/items/comestibles/junkfood.json:1568]:
```suggestion
    "description": "A famous Italian pasta dish made with egg, pecorino and bacon.",
```
Referring to the container won't make sense if the container can be swapped 🤔

File: data/json/items/comestibles/junkfood.json [hunk 17]
```diff
@@ -1551,5 +1551,53 @@
...
+    "weight": "150 g",
+    "color": "red",
+    "spoils_in": "21 days",
+    "container": "pan",
+    "comestible_type": "FOOD",
+    "symbol": "a",
+    "quench": -2,
+    "healthy": -1,
+    "calories": 350,
+    "description": "a pan containing amatriciana pasta, a famous Italian dish made with red sauce, pecorino and bacon.",
```
Comment (TheShadowFerret) [inline data/json/items/comestibles/junkfood.json:1592]:
```suggestion
    "description": "A famous Italian pasta dish made with red sauce, pecorino and bacon.",
```

File: data/json/recipes/food/casseroles.json [hunk 18]
```diff
@@ -121,5 +121,28 @@
...
+  },
+  {
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "eggplant_parmigiana",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_VEGGI",
+    "skill_used": "cooking",
+    "difficulty": 2,
+    "charges": 8,
```
Comment (Maleclypse) [inline data/json/recipes/food/casseroles.json:133]:
```suggestion
    "result_mult": 4,
```

File: data/json/recipes/food/dairy_products.json [hunk 19]
```diff
@@ -192,5 +192,43 @@
...
+  },
+  {
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "tiramisu",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_OTHER",
+    "skill_used": "cooking",
+    "difficulty": 2,
+    "charges": 8,
```
Comment (Maleclypse) [inline data/json/recipes/food/dairy_products.json:204]:
```suggestion
    "result_mult": 4,
```

File: data/json/recipes/food/dairy_products.json [hunk 20]
```diff
@@ -192,5 +192,43 @@
...
+  },
+  {
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "mascarpone",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_OTHER",
+    "skill_used": "cooking",
+    "difficulty": 3,
+    "charges": 6,
```
Comment (Maleclypse) [inline data/json/recipes/food/dairy_products.json:227]:
```suggestion
    "result_mult": 6,
```

File: data/json/recipes/food/pasta.json [hunk 21]
```diff
@@ -222,5 +222,51 @@
...
+  {
+    "result": "pasta_carbonara",
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_PASTA",
+    "skill_used": "cooking",
+    "difficulty": 4,
+    "time": "25 m",
+    "charges": 4,
```
Comment (Maleclypse) [inline data/json/recipes/food/pasta.json:235]:
```suggestion
    "result_mult": 4,
```

File: data/json/recipes/food/pasta.json [hunk 22]
```diff
@@ -222,5 +222,51 @@
...
+  {
+    "result": "amatriciana_pasta",
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_PASTA",
+    "skill_used": "cooking",
+    "difficulty": 3,
+    "time": "25 m",
+    "charges": 4,
```
Comment (Maleclypse) [inline data/json/recipes/food/pasta.json:258]:
```suggestion
    "result_mult": 4,
```

File: data/json/recipes/food/bread.json [hunk 23]
```diff
@@ -301,5 +301,57 @@
     "result": "PBJ_Toast_wheat_free",
     "copy-from": "buttered_toast",
     "components": [ [ [ "jam_fruit", 1 ] ], [ [ "butter", 1 ] ], [ [ "any_peanutbutter", 1, "LIST" ] ], [ [ "toast_wheat_free", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "biscuit",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_BREAD",
```
Comment (Maleclypse) [inline data/json/recipes/food/bread.json:310]: Why did you add two new biscuit recipes that don't have prefixes or suffixes?

Comment (Jackody2) [reply data/json/recipes/food/bread.json:310]: was an error now fixed

**Comment by Maleclypse** [comment]
../tests/comestible_test.cpp:230: FAILED:
(~[slow] ~[.],starting_items)=>   CHECK( lower_bound <= mystats.calories.avg() )
(~[slow] ~[.],starting_items)=> with expansion:
Error: (~[slow] ~[.],starting_items)=>   1803.0f <= 1100.6
(~[slow] ~[.],starting_items)=> 
(~[slow] ~[.],starting_items)=> 
(~[slow] ~[.],starting_items)=> 
(~[slow] ~[.],starting_items)=> RecipeID: eggplant_parmigiana, default is 2404 Calories,
(~[slow] ~[.],starting_items)=> Current recipe range: 976-1265, Average 1100.6, Stddev 78.5
(~[slow] ~[.],starting_items)=> Average recipe Calories must fall within this range, derived from default Calories: 1803-3005

File: data/json/items/comestibles/casseroles.json [hunk 24]
```diff
@@ -74,5 +74,28 @@
...
+    "name": { "str": "eggplant parmigiana" },
+    "weight": "178 g",
+    "color": "brown",
+    "spoils_in": "60 days",
+    "container": "casserole",
+    "comestible_type": "FOOD",
+    "symbol": "p",
+    "quench": -2,
+    "healthy": -1,
+    "calories": 601,
```
Comment (Maleclypse) [inline data/json/items/comestibles/casseroles.json:90]:
```suggestion
    "calories": 301,
```

File: data/json/itemgroups/Food/food.json [hunk 25]
```diff
@@ -338,7 +338,8 @@
       { "item": "biscuit", "prob": 12 },
       { "item": "hardtack", "prob": 2 },
       { "item": "seasoning_salt", "prob": 25 },
-      { "item": "seasoning_italian", "prob": 25 }
+      { "item": "seasoning_italian", "prob": 25 },
+      { "item": "eggplant", "prob": 30 }
```
Comment (Maleclypse) [inline data/json/itemgroups/Food/food.json:342]:
```suggestion
      { "item": "pecorino_cheese", "prob": 30 },
      { "item": "eggplant", "prob": 30 }
```

File: data/json/itemgroups/Food/food.json [hunk 26]
```diff
@@ -670,7 +675,8 @@
       { "item": "pie", "prob": 20 },
       { "item": "pie_veggy", "prob": 15 },
       { "item": "pie_meat", "prob": 15 },
-      { "item": "pie_maple", "prob": 10 }
+      { "item": "pie_maple", "prob": 10 },
```
Comment (Maleclypse) [inline data/json/itemgroups/Food/food.json:678]:
```suggestion
      { "item": "pie_maple", "prob": 10 },
      { "item": "pecorino_cheese", "prob": 10 },
```

**Comment by Sniper27IT** [comment]
I have a suggestion for mascarpone: since the description states that it's made with citric acid and lemons are kinda rarish, it would be wise to add citric acid as alternate ingredient to the 2 lemons. One recipe for citric acid uses 8 lemons for 18 units of citric, so i'd say that 5 units of citric acid could be reasonable for mascarpone.

**Comment by Maleclypse** [comment]
> I have a suggestion for mascarpone: since the description states that it's made with citric acid and lemons are kinda rarish, it would be wise to add citric acid as alternate ingredient to the 2 lemons. One recipe for citric acid uses 8 lemons for 18 units of citric, so i'd say that 5 units of citric acid could be reasonable for mascarpone.

I think if you wanted to PR that it would be accepted.
